package/abs/rancher-abs

```diff
@@ -0,0 +1,464 @@
+#!/bin/bash
+if [ -e "$(dirname $0)/common.sh" ]; then
+    source $(dirname $0)/common.sh
+elif [ -e "$(dirname $0)/../common/common.sh" ]; then
+    source $(dirname $0)/../common/common.sh
+fi
+
+readonly ACCESS_KEY_ID=${ECS_ACCESS_KEY_ID}
+readonly ACCESS_KEY_SECRET=${ECS_ACCESS_KEY_SECRET}
+
+readonly LOOP_CHECK_TIMES=30
+readonly WAIT_SLEEP_TIME_IN_SECONDS=1
+
+TRANSCATION_SEQ=""
+INSTANCE_ID=""
+DISK_ID=""
+
+
+LOG() {
+    if [ "${RANCHER_DEBUG}"x = "true"x ]; then
+        echo -e $1 >> $2 
+    fi
+}
+
+
+# Get which device path in attached disk
+# - # https://help.aliyun.com/document_detail/25514.html?spm=5176.doc25516.6.868.8l8Kr9
+GET_ATTACHED_DEV() {
+    local aliyun_options="--DiskIds '[\"${DISK_ID}\"]'"
+    local aliyun_call_result=$(aliyuncli ecs DescribeDisks "${aliyun_options}" --output json --filter Disks.Disk[0].Description 2>&1)
+    if [ $? -ne 0 ]; then
+        print_error "Failed in DescribeDisks ${DISK_ID}, response ${aliyun_call_result}"
+    fi
+
+    if [ -n "${aliyun_call_result}" ]; then
+        aliyun_call_result=$(echo "${aliyun_call_result}" | sed "s/\"//g")
+    fi
+
+    if [ "$(echo ${aliyun_call_result} | grep 'rancher_abs')" ]; then
+        aliyun_call_result=${aliyun_call_result#'/rancher_abs/'}
+    else
+        aliyun_call_result=""
+    fi
+
+    echo "${aliyun_call_result}"
+}
+
+
+# Gen clientToken
+GEN_CLIENT_TOKEN() {
+    local time_stampe=$(date +%s%N)
+    local random_char=$(cat /dev/urandom | tr -dc 'a-zA-Z' | fold -w5 | head -n1)
+
+    echo "${random_char}_${time_stampe}"
+}
+
+# Gen diskName
```

Comment:
Comments are too obvious. Let code speak for itself. Don't add comments unless necessary.